include/dev/dfi/packet.h

```diff
@@ -572,9 +580,119 @@ wddr_return_t create_cke_packet_sequence(dfi_tx_packet_buffer_t *buffer,
  * @retval      WDDR_ERROR otherwise.
  */
 wddr_return_t create_mrw_packet_sequence(dfi_tx_packet_buffer_t *buffer,
-                                        wddr_freq_ratio_t ratio,
-                                        chipselect_t cs,
-                                        uint8_t mode_register,
-                                        uint8_t op,
-                                        uint16_t time_offset);
+                                         wddr_freq_ratio_t ratio,
+                                         chipselect_t cs,
+                                         uint8_t mode_register,
+                                         uint8_t op,
+                                         uint16_t time_offset);
+
+/**
+ * @brief   Create Packet Group Info
+ *
+ * @details Creates group info required for filling packet with the appropriate
+ *          phases and offsets being accounted for.
+ *
+ * @param[in]   group_info      pointer to group info structure to populate.
+ * @param[in]   ratio           ratio of DFI CK to DRAM CK.
+ * @param[in]   dram_clk_cycles group offset specified in DRAM cycles.
+ * @param[in]   phase_length    duration of group specified in phases.
+ *
+ * @return      void.
+ */
+void create_packet_group_info(packet_group_info_t *group_info,
+                              wddr_freq_ratio_t ratio,
+                              uint16_t dram_clk_cycles,
+                              uint16_t phase_length);
+
+/**
+ * @brief   Fill ADDR Packet
+ *
+ * @details Fills correct WRDATA phases for the specified packet group.
+ *
+ * @param[in]   packet              pointer to packet to fill.
+ * @param[in]   addr_info           pointer to packet group info structure.
```

Comment:
I would prefer if you just call addr_info group_info. The addr_info variable name doesn't make it obvious that it's referring to a packet group. Just my opinion. If you want to keep it, I will still approve.